src/librustc_trans/base.rs

```diff
@@ -578,8 +591,8 @@ fn maybe_create_entry_wrapper(ccx: &CrateContext) {
 
         let (start_fn, args) = if use_start_lang_item {
             let start_def_id = ccx.tcx().require_lang_item(StartFnLangItem);
-            let start_instance = Instance::mono(ccx.tcx(), start_def_id);
-            let start_fn = callee::get_fn(ccx, start_instance);
+            let start_fn = callee::resolve_and_get_fn(ccx, start_def_id, ccx.tcx().mk_substs(
```

Comment:
Because user-defined `start` is supposed to be non-generic, passing a substs to it could cause problems, so please don't.

Reply:
You are right, but user defined `start` is handled in the else branch.